packages/plugin-xml/src/Attributes.ts

```diff
@@ -0,0 +1,45 @@
+import { Modifier } from '../../core/src/Modifier';
+import { deepEqualObjects } from '../../utils/src/utils';
+
+export class Attributes extends Modifier {
+    value: Record<string, string> = {};
+    constructor(valueOrAttributes?: Record<string, string> | Attributes) {
+        super();
+        if (valueOrAttributes instanceof Attributes) {
+            this.value = { ...valueOrAttributes.value };
+        } else if (valueOrAttributes) {
+            this.value = valueOrAttributes;
+        }
+    }
+    get length(): number {
+        return Object.keys(this.value).length;
+    }
+    get name(): string {
+        const name = [];
+        for (const attributeName of Object.keys(this.value)) {
+            name.push(attributeName + ': ' + this.get(attributeName));
+        }
+        return name.join('; ');
+    }
+    get(name: string): string {
+        return this.value[name];
+    }
+    set(name: string, value: string): void {
+        this.value[name] = value;
+    }
+    remove(name: string): void {
+        delete this.value[name];
+    }
+    isSameAs(otherAttributes: Attributes): boolean {
+        if (typeof otherAttributes === 'undefined' && !this.length) return true;
+        return (
+            otherAttributes instanceof this.constructor &&
```

Comment:
What ?

Reply:
Right. I figured, we're making something very extensible, we have classes that extend Attributes, maybe we'll have something else, I don't know, clearly it's mega-overkill (incidentally you'll note that in this instance "mega" in "mega-overkill" is itself overkill).

Reply:
I think if a class extends `Attributes` then probably `Attributes` should consider it to be the same ? Only the other class should know whether it is or not equivalent to a standard `Attributes`.